### ISSUE TITLE
Bugs maintaining instances with multiple volume groups

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#212]
Originally reported of Google Code with ID 156.

```
gnt-cluster (ganeti v2.4.1) 2.4.1

Software version: 2.4.1
Internode protocol: 2040000
Configuration format: 2040000
OS api version: 20
Export interface: 0

Fedora 14 64-bit using KVM

---------------------------------------------

If you create an instance with storage using multiple volume groups, the instance creates and runs just fine, but at least two maintenance functions do not seem to be multiple VG aware and break.

If you run gnt-instance modify '-t plain' => '-t drbd' you get an error running lvrename trying to rename a volume across group names.

You get a similar error with gnt-instance replace-disks.

It appears that both of these functions use the default group name in creating new volumes instead of the group that is being cloned/manipulated/replaced.

This patch seems to fix both bug.  This type of but may exist elsewhere in other functions.

This issue appears to be different than issue 152 which deals with ignoring VGs that ganeti does not "own".

--- cut here ---
--- cmdlib.py0  2011-04-09 22:52:25.118626002 -0400
+++ cmdlib.py   2011-04-09 22:56:24.259626002 -0400
@@ -8260,7 +8260,6 @@
                                  (node_name, self.instance.name))
 
   def _CreateNewStorage(self, node_name):
-    vgname = self.cfg.GetVGName()
     iv_names = {}
 
     for idx, dev in enumerate(self.instance.disks):
@@ -8274,6 +8273,7 @@
       lv_names = [".disk%d_%s" % (idx, suffix) for suffix in ["data", "meta"]]
       names = _GenerateUniqueNames(self.lu, lv_names)
 
+      vgname = dev.children[0].logical_id[0]
       lv_data = objects.Disk(dev_type=constants.LD_LV, size=dev.size,
                              logical_id=(vgname, names[0]))
       lv_meta = objects.Disk(dev_type=constants.LD_LV, size=128,
@@ -9410,7 +9410,7 @@
     snode = self.op.remote_node
 
     # create a fake disk info for _GenerateDiskTemplate
-    disk_info = [{"size": d.size, "mode": d.mode} for d in instance.disks]
+    disk_info = [{"size": d.size, "mode": d.mode, "vg": d.logical_id[0] } for d in instance.disks]
     new_disks = _GenerateDiskTemplate(self, self.op.disk_template,
                                       instance.name, pnode, [snode],
                                       disk_info, None, None, 0, feedback_fn)
--- cut here ---

```
Originally added on 2011-04-11 17:33:12 +0000 UTC.

Mentioned Issue with GoogleCode ID 152 has been migrated to GitHub with ID #208.

**Comment by Ganeti-Issues-Migrator** [comment]
```
This has been fixed and will be in the 2.4.2 release.

Thanks!
```
Originally added on 2011-04-28 11:33:26 +0000 UTC.

Added Labels: Type-Defect
Changed State: Fixed
Added to Milestone: Release2.4

**Comment by Ganeti-Issues-Migrator** [comment]
```
Migrate old issues from 'fixed' to 'released'.
```
Originally added on 2012-10-22 09:16:17 +0000 UTC.

Changed State: Released
